lib/formz.dart

```diff
@@ -38,6 +38,47 @@ extension FormzSubmissionStatusX on FormzSubmissionStatus {
   bool get isCanceled => this == FormzSubmissionStatus.canceled;
 }
 
+/// {@template form_input}
+/// A [FormzInputBase] represents the base class of the [FormzInput] and
+/// [AsyncFormzInput] classes. This class is used to validate by the [Formz]
+/// class its validate method.
+///
+/// ```dart
+/// final FormzInput syncInput = NameInput.dirty(value: 'jan');
+/// final AsyncFormzInput asyncInput = EmailInput.dirty(value: 'test@test.com'),
+///
+/// const inputs = <FormzInputBase>[
+///   FormzInput,
+///   AsyncFormzInput,
+/// ];
+///
+/// final status = Formz.validate(validInputs);
+/// ```
+/// {@endtemplate}
+@immutable
+abstract class FormzInputBase<T, E> {
```

Comment:
This is a nice abstraction from the premise that sync and async inputs would have different classes. That may not necessarily be the case. See [this proposal.](https://github.com/VeryGoodOpenSource/formz/issues/15#issuecomment-1134691056)